### PR TITLE
[Merged by Bors] - Add a small delay after epoch start for sending tortoise beacon messages

### LINKED ISSUE #2528: Flaky `TestMultiNode` and `TestMultipleNodes`
## Description

`TestMultiNode` and `TestMultipleNodes` sometimes fail after merging https://github.com/spacemeshos/go-spacemesh/pull/2267.

## Steps To Reproduce
1. Run `TestMultiNode`/`TestMultipleNodes` until it fails

## Expected Behavior

Both tests always pass

## Environment
Please complete the following information:
Commit `662420dca2690b708899f300712be5fc9a45d9d0`

## Additional Resources 

TBD

### PR DESCRIPTION
**Pull request by nkryuchkov**
## Motivation

Closes #2528 

Epoch start times can differ by a few milliseconds on different nodes. If the tortoise beacon starts sending messages immediately after the epoch start time, some nodes can receive those messages while they are still in a previous epoch, which leads to message verification failures. This PR adds a small configurable delay before the tortoise beacon starts sending messages.

## Changes

- Add a small delay after epoch start for sending tortoise beacon messages

## Test Plan

Manual testing, `TestMultiNode`

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

File: blocks/blockeligibilityvalidator.go [hunk 1]
```diff
@@ -115,9 +116,12 @@ func (v BlockEligibilityValidator) BlockSignedAndEligible(block *types.Block) (b
 	vrfSig := block.EligibilityProof.Sig
 
 	if !v.validateVRF(vrfPubkey, message, vrfSig) {
+		v.log.Error("tortoise beacon eligibility VRF validation of beacon %v in epoch %v failed (counter: %v)", util.Bytes2Hex(epochBeacon), epochNumber, counter)
```
Comment (antonlerner) [inline blocks/blockeligibilityvalidator.go]: make sure the calling function prints out the error, move this format to returned error and you can remove this log line

**Comment by antonlerner** [comment]
what happens when nodes receive messages before the epoch start? maybe use some buffer to store these messages until the epoch starts for this node as well, from our experience although adding some delay helps sometimes, but at other times the delay differs causing these problems to re occur, add this fix, but consider adding more resilience to code itself

**Review by antonlerner — APPROVED** [review]

**Comment by nkryuchkov** [comment]
bors merge

File: blocks/blockeligibilityvalidator.go [hunk 2]
```diff
@@ -102,9 +103,11 @@ func (v BlockEligibilityValidator) BlockSignedAndEligible(block *types.Block) (b
 	vrfSig := block.EligibilityProof.Sig
 
 	if !v.validateVRF(vrfPubkey, message, vrfSig) {
-		return false, fmt.Errorf("eligibility VRF validation failed")
+		return false, fmt.Errorf("tortoise beacon eligibility VRF validation failed: beacon %v, epoch %v, counter: %v", util.Bytes2Hex(epochBeacon), epochNumber, counter)
```
Comment (antonlerner) [inline blocks/blockeligibilityvalidator.go]: `util.Bytes2Hex(epochBeacon)` can be a short string

**Review by antonlerner — APPROVED** [review]

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
bors merge

**Pull request closed without merging**